server/inspect.go

```diff
@@ -254,6 +257,22 @@ func (s *Server) GetExtendInterfaceMux(enableProfile bool) *chi.Mux {
 		}
 	}))
 
+	mux.Get(InspectHeapEndpoint, http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		w.Header().Set("Content-Type", "text/plain")
+
+		f, err := os.CreateTemp("", "cri-o-heap-*.out")
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		debug.WriteHeapDump(f.Fd())
+
+		if _, err := fmt.Fprintf(w, "Wrote heap dump to: %s\n", f.Name()); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}))
```

Comment:
Now that this is more user-facing, so to speak, having a slightly different approach could be more prudent.

A future CLI could use one of the following approaches:

- `crio status heap` - without any argument, it will write the heap dump to local/current directory rather than the `/tmp` (so not quite like what sending the signal currently does)
- `crio status heap` - without argument simply writes to the standard output (but this might be too messy)
- `crio status heap -f <FILE NAME>` - the `-f` or `--file` (or perhaps even `--output`) will be a required argument denoting where the heap dump should be written
- `crio status heap -f -` will write the heap dump to the standard output (since this will be done on request, the user has to handle the output)

Reply:
You mean "heap.out". :smile: 

Looking at...

https://github.com/cri-o/cri-o/blob/7a9778cadc77302b17de7d637b28d1c8089313a3/cmd/crio/main.go#L43-L52

If we extract the date bit into a small helper (to reuse the compressed timestamp), it could then help us to render a file name automatically following the pattern we already have for the SIGUSR1 handling. So, we pick one:

- crio-heap-%s.out
- crio-heap-dump-%s.out
- crio-heap-%s.dump

Add the compressed timestamp, and et voila.

Reply:
Okay, changed as suggested :+1: